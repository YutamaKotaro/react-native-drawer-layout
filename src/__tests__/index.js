import { Text } from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import DrawerLayout from '../DrawerLayout';

describe('DrawerLayout', () => {
  it('renders closed correctly', () => {
    const tree = renderer
      .create(
        <DrawerLayout
          drawerWidth={300}
          drawerPosition={DrawerLayout.positions.Left}
          renderNavigationView={() => <Text>Drawer</Text>}
        >
          <Text>Content!</Text>
        </DrawerLayout>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
