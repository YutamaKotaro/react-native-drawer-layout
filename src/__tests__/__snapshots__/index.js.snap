// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerLayout renders closed correctly 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Content!
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": undefined,
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "translateX": -300,
          },
        ],
        "width": 300,
        "zIndex": 1001,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Drawer
    </Text>
  </View>
</View>
`;
